cmd/operator/main.go

```diff
@@ -191,7 +209,12 @@ func getLogger(logArgs Logging) *zap.Logger {
 		zapcore.NewMultiWriteSyncer(writes...),
 		lvl,
 	)
-	return zap.New(core)
+	opts = append(opts, getStackTrace(logArgs.DevMode))
+	if !logArgs.DevMode {
+		// This enables sampling only in prod
+		core = zapcore.NewSamplerWithOptions(core, time.Second, First, ThereAfter)
```

Comment:
I don't know much about sampling so those values were taken from  [NewProductionConfig()](https://github.com/uber-go/zap/blob/feeb9a050b31b40eec6f2470e7599eeeadfe5bdd/config.go#L119). How important is sampling?

Reply:
Why can't we use `NewProductionConfig`?  Sampling will help reduce IO and CPU cost of the logger.  So I think its something that we want.

Reply:
To create a zap.Logger there are 3 options: 

1. using  `NewProductionConfig/NewDevelopmentConfig` + build() or simply `NewProduction/NewDevelopment` if you don't want to configure manually. This is quite easy as most of the configuration has already been done but it is less flexible.
2. defining a Config struct and then do all the configuration manually. This is used when you really wants to configure at a deep level. functions mentionned above used a Config struct. Then use `Build()` method to create a logger.
3. using `New(core zapcore.Core,options ...Option)` to create a logger from a `zapcore.Core.` This is the most flexible but some configuration are not done automatically (like stack trace) and you need to add them through options.

There is not one thing you can do with one option that you cannot with the others but of Course the level of difficulty is different. In our case we are using lumberjack so the better option is the 3rd as recommended by zap. The use of lumberjack is what makes me choose the 3rd.

I think there is still a way to use NewProductionConfig by using some of these [functions](https://pkg.go.dev/go.uber.org/zap@v1.15.0#Option) to pass the zapcore.Core as option to the `Build()` method but this will require to define some global variables and to change some functions. So basically replacing/altering the current Core. I think it is more complicated and the code is less organized.

Reply:
Thanks for the detailed explanation.  I agree with your approach of using option 3.

Back to your original question, I think taking the sampling settings you copied from `NewProductionConfig` makes the most sense.